core/templates/dev/head/domain/exploration/WrittenTranslationObjectFactory.ts

```diff
@@ -17,53 +17,52 @@
  * WrittenTranslation domain objects.
  */
 
-var oppia = require('AppInit.ts').module;
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-oppia.factory('WrittenTranslationObjectFactory', [function() {
-  var WrittenTranslation = function(html, needsUpdate) {
+export class WrittenTranslation {
+  html: any;
+  needsUpdate: any;
+  constructor(html: any, needsUpdate: any) {
     this.html = html;
     this.needsUpdate = needsUpdate;
-  };
-
-  WrittenTranslation.prototype.getHtml = function() {
+  }
+  getHtml() {
     return this.html;
-  };
-
-  WrittenTranslation.prototype.setHtml = function(html) {
+  }
+  setHtml(html: any) {
     this.html = html;
-  };
-
-  WrittenTranslation.prototype.markAsNeedingUpdate = function() {
+  }
+  markAsNeedingUpdate() {
     this.needsUpdate = true;
-  };
-
-  WrittenTranslation.prototype.toggleNeedsUpdateAttribute = function() {
+  }
+  toggleNeedsUpdateAttribute() {
     this.needsUpdate = !this.needsUpdate;
-  };
-
-  WrittenTranslation.prototype.toBackendDict = function() {
+  }
+  toBackendDict() {
     return {
       html: this.html,
       needs_update: this.needsUpdate
     };
-  };
+  }
+}
 
-  // TODO (ankita240796) Remove the bracket notation once Angular2 gets in.
-  /* eslint-disable dot-notation */
-  WrittenTranslation['createNew'] = function(html) {
-  /* eslint-enable dot-notation */
+@Injectable({
+  providedIn: 'root'
+})
+export class WrittenTranslationObjectFactory {
+  createNew(html) {
```

Comment:
Declare `any` type.

Reply:
Should we be doing this? Since `any` is already implied?

Reply:
Since we're already making explicit declarations elsewhere, we should continue that convention, otherwise we'll be mixing conventions.

Reply:
Done and added more strong typing elsewhere too.